### ISSUE TITLE
return inspect() does not return a value but assigning to a variable does.

### ISSUE THREAD

**Issue by idiv0** [bcardosolopes/llvm-project#7]
When trying to return the value from a call to `inspect` the compiler emits the following warning:
`<source>:9:1: warning: non-void function does not return a value [-Wreturn-type]`

The code prints the correct value despite the warning.
Sample code below, and Godbolt link: https://godbolt.org/z/q3xr33
```
#include <iostream>
#include <utility>

int factorial(int n) {
    return inspect(n) {
        0 => 1;
        __ => n * factorial(n-1);
    };
}

int main() {
  std::cout << factorial(5) << "\n";
}
```

The same logical code does not produce a warning when assigning the output of inspect into a local variable and then returning that.
Working Godbolt example: https://godbolt.org/z/MPh5s4

```
#include <iostream>
#include <utility>

int factorial(int n) {
    auto value = inspect(n) {
        0 => 1;
        __ => n * factorial(n-1);
    };
    return value;
}

int main() {
  std::cout << factorial(4) << "\n";
}
```

**Comment by dansarginson** [comment]
This looks like a lack of CFG to me. This is just a [stub](https://github.com/bcardosolopes/llvm-project/blob/llvmorg-master-pattern-matching/clang/lib/Analysis/CFG.cpp#L4272) for inspect. We had a plan to tackle this quite soon, so watch this space :-)

**Comment by bcardosolopes** [comment]
This is right, stay tuned, this will land soon!

**Comment by bcardosolopes** [comment]
Fixed in 390624920a98dc93a5273245d0a9375cc85473b5
